### PR TITLE
Allow untested maximum versions in our Python libraries

### PR DESCRIPTION
**Pull request by willbarton**
@m3brown has proposed (in https://github.com/cfpb/wagtail-inventory/pull/36 and https://github.com/cfpb/wagtail-inventory/issues/37) that we lift the requirement for a maximum version that is tested in our Python libraries. This would allow for existing releases of our libraries to support new minor and patch releases of our dependencies that don't break compatibility.

We currently specify our major dependencies like:

```python
install_requires = [
    "django>=3.1,<3.2",
    "wagtail>=2.7,<2.11",
]
```

Where Django 3.2 and Wagtail 2.11 were not yet released at the time, but Django 3.1 and Wagtail 2.10 were.

@m3brown proposed allowing something like:
```python
install_requires = [
    "django>=3.1",
    "wagtail>=2.7",
]
```

Which this proposed change in language wouldn't allow. But it would allow us to specify:

```python
install_requires = [
    "django>=3.1,<4.0",
    "wagtail>=2.7,<3.0",
]
```

Which seems reasonable. There's a big caveat with Wagtail though, as @chosak [points out](https://github.com/cfpb/wagtail-inventory/pull/36#issuecomment-721821326), Wagtail doesn't always strictly follow semantic versioning, and minor releases do break compatibility.

This is my proposal for new language that allows that, submitted for discussion. It also cleans up a long run-on sentence.

[Here is the change rendered](https://github.com/cfpb/development/blob/allow-nebulous-max-versions/standards/python.md#specifying-dependencies-in-requirementstxt-or-setuppy).

### CODE REVIEW

File: standards/python.md [hunk 1]
```diff
@@ -60,4 +60,4 @@ The difference between pinned versions of dependencies in `requirements.txt`  an
...
+Our libraries should specify dependencies with supported version ranges. The minimum version in the range and the last available version [should be tested with tox](../tox.ini) and can be major, minor, or match). The maximum version that is expected to break compatibility can be a theoretical future major major. The range should be specified in `setup.py` using [the appropriate setuptools `_requires` keyword](https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-dependencies). For example, our [wagtail-sharing library](https://github.com/cfpb/wagtail-sharing) declares install-time requirements with `install_requires` and testing requirements with `extras_require={'testing': testing_extras}`.
```
Comment (chosak) [inline standards/python.md:63]: What does ["major major"](https://en.wikipedia.org/wiki/Major_Major_Major_Major) mean here?

Also, it looks like the documentation link for setuptools dependencies [has moved](https://setuptools.readthedocs.io/en/latest/userguide/dependency_management.html).

Comment (higs4281) [reply standards/python.md:63]: > major, minor, or match ...

patch?

**Comment by schbetsy** [comment]
Hypothetically, if we did allow flexible library versions in our repos, and one of them did introduce a breaking change, where would that get caught?

I think your suggestion could work, as long as we consistently test that the latest allowed version actually works. Could we add an automated test that always pulls the latest versions of any libraries with flexible versions? That way we could be notified if there is a breaking change so we'd know to change pinned versions or fix new bugs.

We discussed this in our Dev Community of Practice meeting, but I wanted to put it here as well.

**Comment by willbarton** [comment]
It's an excellent idea, and we've tried doing so in wagtail-flags: https://github.com/cfpb/wagtail-flags/pull/61

I think that might work. 

I'll clean up the silly copy mistakes noted above and then I think this might be ready for approval?

**Comment by willbarton** [comment]
Alright, I think this is ready for review again!

**Comment by willbarton** [comment]
I should add, this approach (with @schbetsy's test environment suggestion) is the approach we've taken with the latest releases of our Wagtail libraries to add support for Wagtail 2.11.

**Review by schbetsy — APPROVED** [review]

**Review by higs4281 — APPROVED** [review]

**Pull request merged**